File: blocking_vs_nonblocking/Demo1.java
class Demo1 {
    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();

        System.out.println("Starting blocking operation...");
        String result = performBlockingOperation();
        System.out.println("Blocking operation result: " + result);

        long endTime = System.currentTimeMillis();
        System.out.println("Time taken: " + (endTime - startTime) + " ms");
    }

    private static String performBlockingOperation() {
        try {
            // Simulate time-consuming I/O operation
            Thread.sleep(3000); // 3 seconds
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        return "Operation completed";
    }
}

File: blocking_vs_nonblocking/Demo1.js
const startTime = new Date();

console.log("Starting blocking operation...");
const reslut = performBlockingOperation();
console.log(`blocking operation result: ${reslut}`);

const endTime = new Date();
console.log(`Time taken: ${endTime - startTime} ms`);

function performBlockingOperation() {
    const start = Date.now();
    while (Date.now() - start < 3000) {
        // Busy-wait for 3 seconds
    }

    return "Operation completed";
}
